### PR TITLE
Struct tag linter

### PR DESCRIPTION
**Pull request by cpuguy83**
Example (this is from a commit I added temporarily: https://github.com/Azure/dalec/commit/5bc521734d090c919ea902137dd332fbfd35b091).

image

### CODE REVIEW

File: linters/yaml_json_tags.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		json = "json"
+	)
+
+	tag = strings.Trim(tag, "`")
+
+	var out [2]string
+	for tag != "" {
+		var key, value string
+		key, tag, _ = strings.Cut(tag, ":")
+		value, tag, _ = strings.Cut(tag, " ")
```
Comment (adamperlin) [inline linters/yaml_json_tags.go:64]: Can you please add a test case for this function and make sure it is fully robust to whitespace between `key:value` pairs?

**Review by adamperlin — CHANGES_REQUESTED** [review]

**Review by adamperlin — APPROVED** [review]
Looks good to me pending the unrelated CI failures!
